src/native/corehost/hostmisc/utils.cpp

```diff
@@ -348,14 +348,25 @@ bool try_stou(const pal::string_t& str, unsigned* num)
     return true;
 }
 
-pal::string_t get_dotnet_root_env_var_name()
+bool get_dotnet_root_from_env(pal::string_t* dotnet_root_env_var_name, pal::string_t* recv)
 {
+    *dotnet_root_env_var_name = _X("DOTNET_ROOT_");
+    dotnet_root_env_var_name->append(to_upper(get_arch()));
+    if (pal::getenv(dotnet_root_env_var_name->c_str(), recv))
+        return pal::file_exists(*recv);
```

Comment:
This is no longer going to resolve/assign the real path to `recv` (just the direct environment variable value).

Reply:
Won't the `realpath` call inside `file_exists` set it for us?

Reply:
On Unix, `file_exists` doesn't call `realpath`. On Windows, it does, but throws away the updated path.